integrated_channels/xapi/serializers.py

```diff
@@ -4,3 +4,56 @@
 Serializers for xAPI data.
 """
 from __future__ import absolute_import, unicode_literals
+
+from rest_framework import serializers
+
+from enterprise.models import EnterpriseCustomerUser
+
+
+class LearnerInfoSerializer(serializers.Serializer):
+    """
+    Serializer for Learner info.
+
+    Model: User
+    """
+    lms_user_id = serializers.IntegerField(source='id')
+    user_username = serializers.CharField(source='username')
+    user_email = serializers.EmailField(source='email')
+    user_country_code = serializers.CharField(source='profile.country.code')
+    user_account_creation_date = serializers.DateTimeField(source='date_joined')
+
+    enterprise_user_id = serializers.SerializerMethodField()
+    enterprise_sso_uid = serializers.SerializerMethodField()
+
+    def get_enterprise_user_id(self, obj):
+        try:
+            enterprise_learner = EnterpriseCustomerUser.objects.get(user_id=obj.id)
+        except EnterpriseCustomerUser.DoesNotExist:
+            return
+
+        return enterprise_learner.id
+
+    def get_enterprise_sso_uid(self, obj):
+        try:
+            enterprise_learner = EnterpriseCustomerUser.objects.get(user_id=obj.id)
+        except EnterpriseCustomerUser.DoesNotExist:
+            return
+
+        return enterprise_learner.enterprise_customer.identity_provider
+
+
+class CourseInfoSerializer(serializers.Serializer):
+    """
+    Serializer for course info.
+
+    Model: CourseOverview
+    """
+    course_id = serializers.CharField(source='id')
+    course_title = serializers.CharField(source='display_name')
+    course_description = serializers.CharField(source='short_description')
+    course_details_url = serializers.CharField(source='marketing_url')
```

Comment:
would we ever want to send the marketing url, since that will point to the B2C landing pages?

Reply:
According to my understanding, the course-details page is the same for both B2B and B2C scenarios. This data will be sent as an extension to X-API object field that contains course details and will only be used by enterprise customer trying to get more info about a course. I think marketing URL is better than enterprise landing page. Let know your thoughts?

Reply:
I think the marketing page for now makes the most sense. Maybe someday we'll have b2b marketing pages in the portal, but for now this is the best place to get course details.